contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsFetcher.java

```diff
@@ -220,6 +217,12 @@ private File fetchFromSource(String sourceFileUrl,
         ObjectName jmxName = null;
         HdfsCopyStats stats = null;
         FileSystem fs = null;
+
+        if (voldemortConfig.isBouncyCastleEnabled()) {
+            //Use bouncy castle provider as first choice. If provider is existed, it will not be inserted again.
```

Comment:
Can you clarify, what do you mean when you say "If provider is existed"? Perhaps "If the BouncyCastleProvider was already inserted"?